lib/mongo/database/view.rb

```diff
@@ -164,8 +180,12 @@ def collections_info_spec(session, options = {})
           db_name: @database.name,
           session: session
         }.tap do |spec|
-          spec[:selector][:nameOnly] = true if options[:name_only]
+          # nameOnly and authorizedCollections default to false
+          if options[:name_only] ? (spec[:selector][:nameOnly] = true) : (spec[:selector][:nameOnly] = false)
+          end
           spec[:selector][:filter] = options[:filter] if options[:filter]
+          if options[:authorized_collections] ? (spec[:selector][:authorizedCollections] = true) : (spec[:selector][:authorizedCollections] = false)
```

Comment:
Similarly here you should be able to just do `          spec[:selector][:authorizedCollections] = true if options[:authorized_collections]`.